### DISCUSSION TITLE
Sensor Failed 7

### DISCUSSION THREAD

**Discussion by Schizm1** [NightscoutFoundation/xDrip#3463] [Q&A] [answered]
It seems like a roll of the dice anytime I start a new G7 sensor. I went through all the instructions on the github page, scanned the auto settings and went through the manual settings. I get asked to "Now start a new sensor" every time and it defaults to the 90 minute timer. Is this right or do I ignore it? If I'm lucky it eventually warms up in the 30 minutes. I just put a new one in, and about 25 minutes in I got a "Sensor Failed". Now at the 46 minute left mark I get a "Sensor Failed 7". I am leaving for a trip tomorrow and I'm stressing. I had one extra sensor and if I have to ditch this one I'm cutting it close.

**Comment by Navid200** [comment]
These are the instructions for using G7 as a collector: https://navid200.github.io/xDrip/docs/Dexcom/G7.html

Can you post a screenshot of your G5/G6/G7/1 Status page?

As an alternative, do you have a Dexcom G7 receiver?
Or, do you have a phone that is compatible with G7 Dexcom app?

**Comment by Schizm1** [comment]
I followed those instructions and triple checked them.
I may have to switch to the dexcom App but is it to late ?

**Reply by Navid200** [reply]
Do what you need to.  I asked you to show the status page.  But, of course you don't have to.  But, why did you post here then?

**Comment by Schizm1** [comment]
That screen is called the System Status so I was under the impression
that's what you were looking for. Do you want the Event Log?

**Reply by Navid200** [reply]
I see nothing.  You need to log into GitHub and go to the discussions and attach an image.  You cannot post here by responding to an email and include an image.  the image will not make it to GitHub.
You can see yourself if you log in.

**Comment by Schizm1** [comment]
![Screenshot_20240514_205253_xDrip+](https://github.com/NightscoutFoundation/xDrip/assets/25514585/79f10a2d-a9bd-488b-87bc-3d077260d362)
![Screenshot_20240514_203016_xDrip+](https://github.com/NightscoutFoundation/xDrip/assets/25514585/2d01f435-d4d2-4a4c-b4b5-1b5b6b14b1a8)
![Screenshot_20240514_202919_xDrip+](https://github.com/NightscoutFoundation/xDrip/assets/25514585/e91b52ee-bc2d-4130-8eca-0718de03b3d6)
![Screenshot_20240514_202936_Settings](https://github.com/NightscoutFoundation/xDrip/assets/25514585/3a7e302b-ed1c-4e67-a9c0-163471cf87c4)

**Reply by Navid200** [reply]
OK

Please install this release as a first step:
https://github.com/NightscoutFoundation/xDrip/releases/tag/2024.03.27

Then, wait 5 minutes.  Then, look under the main menu (top left).  Do you see a start sensor option?  If yes, start.

Tell me what happens.

**Reply by Navid200** [reply]
And can I ask how long ago you got this transmitter?  Voltage B is unusually low.

**Reply by Schizm1** [reply]
Just installed. Its a brand new transmitter. I get them at 3 month supply and this is the 3rd to last.

**Reply by Navid200** [reply]
OK

What does status page show now?

**Comment by Schizm1** [comment]
No Start Sensor option.
![Screenshot_20240514_212011_xDrip+](https://github.com/NightscoutFoundation/xDrip/assets/25514585/f3d28533-0915-402f-8f7e-0dce88c50f85)
![Screenshot_20240514_212019_xDrip+](https://github.com/NightscoutFoundation/xDrip/assets/25514585/10632430-ffc9-4832-9118-0a8af79a7c75)

**Reply by Navid200** [reply]
Have you disabled trust auto-connect under the Bluetooth settings?

Have you updated your phone to Android 14?
Have you installed the latest available patches from Samsung?

**Comment by Schizm1** [comment]
Auto-Connect disabled (for Samsung)
Android Version 14
Never have installed patches (Verizon)?

**Reply by Navid200** [reply]
Edit the transmitter ID and change the last character to something else and apply.
Go to your phone Bluetooth settings and find the list of previously or currently connected devices.  Delete (forget) all starting with DXCM or DX02.

Put all your old used G7 devices in a metal box or microwave or very far away.

Now, change the transmitter ID back to the correct value and go to the status page and wait for a pair request and approve.

**Comment by Schizm1** [comment]
My phone updated this afternoon S906USQU4EXE1

**Comment by Schizm1** [comment]
Ok re-paired 
![Screenshot_20240514_214640_xDrip+](https://github.com/NightscoutFoundation/xDrip/assets/25514585/b8e6d8a0-af41-4d0b-a8b5-0073331db69c)

**Reply by Navid200** [reply]
Wait for another read cycle.
Is your blood glucose low?  If yes, treat yourself.

**Comment by Schizm1** [comment]
Bg level is 154

![Screenshot_20240514_215557_xDrip+](https://github.com/NightscoutFoundation/xDrip/assets/25514585/d58a5a9c-2662-42e4-a9c1-2d9f9aa16289)

**Reply by Navid200** [reply]
Looking at the sensor, can you see the filament partly visible from the top?

**Comment by Schizm1** [comment]
Just a tiny bit, very slightly

**Reply by Navid200** [reply]
Can I see a picture?

**Reply by Navid200** [reply]
Is it like the third picture here?
https://navid200.github.io/xDrip/docs/Dexcom/G7.html

**Reply by Schizm1** [reply]
It didn't seem to look like the one in the picture at all but maybe it is?

![20240514_220802](https://github.com/NightscoutFoundation/xDrip/assets/25514585/1a537830-0dca-431d-982b-29e5fb3524ca)

**Reply by Navid200** [reply]
Yes, that is an applicator failure.
May be you can ask someone to help you take a better picture with better focus to show Dexcom if they ask.

There is no way this one would work regardless of which app you use.
Contact Dexcom and they will replace it.

**Reply by Schizm1** [reply]
Well, that stinks. Sorry if that's the case, I cant see it very well and definetely can't feel it. If I put in another sensor, is it normal for Xdrip to say "Start Sensor" with a G7 and start a 90 min countdown? G7s are only supposed to take 30 min.

**Reply by Navid200** [reply]
No, it should not.  If you follow the instructions, it will not ask you to start anything.
Take a screenshot and show me the next time you see it.

After you establish connectivity to a sensor that you have already applied, xDrip will first say now you have to wait for connectivity.  Then, it will say warmup is in progress.

**Reply by Schizm1** [reply]
Ok, thanks. I always get the red message at the top of the main screen that states "Now start your sensor". Should I just ignore it if I do see it this time after connection?

**Reply by Navid200** [reply, marked as answer]
You were using an older version of xDrip.
We have updated a lot for the sake of G7.  But, we only release a stable release once every 6 months.  You were using a release that did not have any of the recent updates.
You will not see that.
But, if you see a message that asks you to start, please show me and yes, ignore it.

**Reply by Schizm1** [reply]
Ok, thank you for your help and I apologize for your time.

**Reply by Navid200** [reply]
No worries.  I am glad we figured out what was going on.

**Comment by Navid200** [comment]
After (and including) the 2024.08.07 xDrip [release](https://github.com/NightscoutFoundation/xDrip/releases/tag/2024.08.07), sensor failed 7 condition will provide readings if readings are available.

**Comment by PieterGit** [comment]
@Navid200 thanks for the update. We upgraded from G6 to G7 today and ran into a "Sensor Failed 7" issue with a new G7 sensor. The sensor was started with Dexcom G7 native app, but that failed after a while (did not provide readings) and asked to insert a new sensor. I removed the Dexcom G7 app and installed/upgraded to Xdrip 2024.08.07 nightly hoping that would fix the data readings. Perhaps the older Xdrip and Dexcom G7 had bad interaction together. The upgrade Xdrip nightly (without Dexcom G7 installed) shows this status:
![image](https://github.com/user-attachments/assets/89e518e0-fb27-44d5-840a-3395be69ceba)

I checked if I can get the battery status, and that seems to work just fine.
I don't seem to get readings, and waited more than an hour.

Notice in the homescreen shows a green arrow with: "22:57 Started by transmitter" and below in the screen you see this:
![image](https://github.com/user-attachments/assets/f1a2155b-bb95-48f3-bf10-ce575e49f4d2).

I think the relevant logfile is (some debug stuff redacted to xxx and changed mac-address ):
```
3 ^ 2024-08-08 23:36:03 ^ G5CollectionService ^ Encountered backfill data we don't recognise: 25 216 3  2024-08-08 20:25:53
3 ^ 2024-08-08 23:31:02 ^ G5CollectionService ^ Encountered backfill data we don't recognise: 25 216 3  2024-08-08 20:25:54
3 ^ 2024-08-08 23:26:02 ^ G5CollectionService ^ Encountered backfill data we don't recognise: 25 216 3  2024-08-08 20:25:54
3 ^ 2024-08-08 23:21:03 ^ G5CollectionService ^ Encountered backfill data we don't recognise: 25 216 3  2024-08-08 20:25:53
3 ^ 2024-08-08 23:16:03 ^ G5CollectionService ^ Encountered backfill data we don't recognise: 25 216 3  2024-08-08 20:25:53
3 ^ 2024-08-08 23:11:03 ^ G5CollectionService ^ Encountered backfill data we don't recognise: 25 216 3  2024-08-08 20:25:54
5 ^ 2024-08-08 23:10:59 ^ Ob1G5StateMachine ^ Saving battery data: Status: OK / VoltageA: 287 / VoltageB: 263 / Resistance: 7680 / Run Time: 0 / Temperature: 0
2 ^ 2024-08-08 23:10:59 ^ Ob1G5StateMachine ^ Store: BatteryRX dbg: xxx
2 ^ 2024-08-08 23:10:59 ^ G5CollectionService ^ BatteryInfoTx dbg: xxx
2 ^ 2024-08-08 23:05:59 ^ Ob1G5StateMachine ^ Wrote version request: 2
2 ^ 2024-08-08 23:05:59 ^ Ob1G5StateMachine ^ Store: VersionRX dbg: xxx
2 ^ 2024-08-08 22:57:11 ^ Ob1G5StateMachine ^ Store: VersionRX dbg: xxx
2 ^ 2024-08-08 22:57:11 ^ Ob1G5StateMachine ^ Wrote version request: 1
2 ^ 2024-08-08 22:57:11 ^ DexTimeKeeper ^ Refusing to update stored timestamp due to excessive processing latency: 14722ms
2 ^ 2024-08-08 22:57:11 ^ Ob1G5StateMachine ^ Cannot process backfill evaluation as no active sensor
6 ^ 2024-08-08 22:57:11 ^ Ob1G5CollectionService ^ Native Sensor is now Started: Sensor Failed 7
2 ^ 2024-08-08 22:57:10 ^ Ob1G5CollectionService ^ onReceive UPDATE Name DXCxxx Value AA:BB:CC:DD:EE:FF Bond state 12 Paired bs:  Paired was  Pairing
5 ^ 2024-08-08 22:57:08 ^ Ob1G5CollectionService ^ Prompting user to notice pairing request with sound - On Android 8+ you have to manually pair when requested
2 ^ 2024-08-08 22:57:08 ^ Ob1G5CollectionService ^ onReceive UPDATE Name DXCxxx Value AA:BB:CC:DD:EE:FF Bond state 11 Pairing bs:  Pairing was  Unpaired
```

If I'm interpreting the logs write it has a problem dat with this sensor Xdrip is not able to process the backfill data.
I don't have an option to calibrate the sensor, so it seems Xdrip is not thinking the sensor has started. 
If I need to check more logging, please let me know what I should search for.
Hope this helps fixing the issue.

**Reply by Navid200** [reply]
There is no reason to blackout the firmware version.  And that is a very useful piece of information for us.  
You can disable collection with xDrip and use Dexcom app.  Does it work with Dexcom app?

**Reply by PieterGit** [reply]
Removed the Xdrip to be sure it is not interacting with Dexcom app. Reinstalled the Dexcom G7 app and after re-entered transmitter number, I got the error that is on page 115 of https://dexcompdf.s3.us-west-2.amazonaws.com/en-us/G7-CGM-Users-Guide.pdf see screenshot below:
![image](https://github.com/user-attachments/assets/d82c514a-31db-41e4-a7a1-d2489d7714cc).

So I assume this sensor really didn't start ok because it give up after only a few hours. It does not give glucose data and Dexcom app stays in the "Start new sensor" state. This was a sensor with firmware  32.192.105.64.

**Reply by Navid200** [reply]
One of the things you should check right after applying the sensor is to gently rub your finger over the sensor to see if you can feel the filament.
If you can as shown on the third image on the following page, there is no way it will ever work.
https://navid200.github.io/xDrip/docs/Dexcom/G7ApplicatorFailures.html  

I have no way of knowing this is what happened.  But, it happens or it used to happen.  Hopefully, Dexcom has fixed that already.
